### PR TITLE
Fix #382

### LINKED ISSUE #382: Cast datetime bug.
```
tidb> select cast("xxx 10:10:10" as datetime);
+----------------------------------+
| CAST("xxx 10:10:10" AS datetime) |
+----------------------------------+
| 2010-10-10 00:00:00              |
+----------------------------------+
```

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
```
tidb> select cast("xxx 10:10:10" as datetime);
+----------------------------------+
| CAST("xxx 10:10:10" AS datetime) |
+----------------------------------+
| 2010-10-10 00:00:00              |
+----------------------------------+
```

See: https://github.com/pingcap/tidb/issues/382

### CODE REVIEW

**Comment by zimulala** [comment]
LGTM

File: mysql/time.go [hunk 1]
```diff
@@ -310,14 +310,41 @@ func (t Time) RoundFrac(fsp int) (Time, error) {
...
+		// Date fromat must start and end with number.
+		if i == 0 || i == len(format)-1 {
+			if !unicode.IsNumber(rune(format[i])) {
+				return nil
+			}
+
+			continue
+		}
+
+		// Sep can be only one none-number char.
```
Comment (shenli) [inline mysql/time.go]: Seperator is a single none-number char.

Comment (qiuyesuifeng) [inline mysql/time.go]: Gotcha.

**Comment by qiuyesuifeng** [comment]
@shenli  PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**